### PR TITLE
Conquest Factory

### PR DESCRIPTION
**Pull request by PhABC**
Simple contract for Conquest mode, where players need to burn a silver card to enter and can win random silver cards depending on their performance. Randomness is done off-chain and server mints the cards. Later we can make this more trustless, but since user enter with silver cards (about 2$), the risk is low.

### CODE REVIEW

File: contracts/shop/SilverConquestFactory.sol [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  |__________________________________*/
+
+  /**
+   * @notice Create factory, link skyweaver assets and store initial parameters
+   * @param _assetsAddr  The address of the ERC-1155 Assets Token contract
+   */
+  constructor(address _assetsAddr) public {
+    require(_assetsAddr != address(0), "SilverConquestFactory#constructor: INVALID_INPUT");
+    skyweaverAssets = ISkyweaverAssets(_assetsAddr);
+    emit MintBurnRatioChange(0, 875);
```
Comment (Agusx1211) [inline contracts/shop/SilverConquestFactory.sol]: Maybe should emit the value on `mintBurnRatio`

**Review by Agusx1211 — APPROVED** [review]

**Comment by Agusx1211** [comment]
LQTM, maybe a future version could use a hash-chain for randomness, to avoid the owner having total control over which ids are minted

**Comment by PhABC** [comment]
> LQTM, maybe a future version could use a hash-chain for randomness, to avoid the owner having total control over which ids are minted

Agreed. Just wanted to minimize the cost and complexity for now so we can test that feature without too much risk.

File: contracts/shop/SilverConquestFactory.sol (hunk 1 above)
Comment (PhABC) [reply contracts/shop/SilverConquestFactory.sol]: Might as well indeed

**Pull request merged**